### DISCUSSION TITLE
Auxillary Fan issue with control logic

### DISCUSSION THREAD

**Discussion by MakManIfest** [Guilouz/Creality-Helper-Script-Wiki#511] [Issues]
The Auxillary fan in My K1 Max has not been obeying the settings in Orca slicer. It was set to 70% by default but runs at 91% during the print.

Proceeded some digging into the cfg's and did some math. Discovered that the way the M106 P2 S(x) gcode is programmed to be responded to by the K1 klipper config is by using the narrow band of values between the minimum setting of 180 as described in [gcode_macro PRINTER_PARAM] in the gcode_macro.cfg, to the max setting of 255. So the 0% to 100% setting in Orca for the Auxillary fan control is ajusting between 180 to 255.

i.e if the fan is set to 70% in Orca the fan will run at 91%. If the fan is set to 80% in Orca the fan will run at 94%. 32% in Orca results in 80% on the printer.

Not sure what can be done about this as the fans minimum run input is 180 which translates to  about 71%. Fan doesn't run very well below 80% in reality.

Anyway thought I'd bring it up. Adjustments might need to be made in Orca to limit the minimum % to 71 to match what is in Klipper. Or adjust the klipper code to set the fan to minimum speed at any value from the M106 P2 command under 71% and accurately set the fan above that.

**Comment by kisslorand** [comment]
It's the `M106` macro that maps the values from 180 to 255.

**Comment by MakManIfest** [comment]
No. Well maybe. Depends on what stuff you have installed through helper-script. But in my config, no.
The M106 macro is commented out, I already checked that.
Various macros that activate the Auxiliary fan reference a variable defined in the printer param gocode macro. This variable defines the minimum speed of the fan.
To directly control the fan, the SET_PIN PIN=fan2 VALUE=xxx is used. xxx being 0-255. This results in the fan being set accurately from 0-100%.
Using the M106 P2 command as out put via Orca results in the behavior described in my OP.

**Comment by Scootworks** [comment]
I'm running with the same issue on my K1C. Also, the exhaust fan doesn't work at all, even if I activate it in OrcaSlicer. I don't know if that is related to each other

**Comment by ZZZXXX0110** [comment]
Noticed the same behavior on my K1C too. I also noticed that the fan speed setting using Creality's UI on the printer's own screen, as well as Creality web interface, all correspond to the % between 180 to 255 like OP describes.

**Comment by shirgall** [comment]
I have run into this one. After a filament runout and reload, side fan did not come back to speed. Setting it from Orca slicer/Klipper only worked for rear fan and part fan. Setting it manually from the console produced random results and eventually a fan control error.

**Comment by ronoaldo** [comment]
I'm observing a weird behavior as well. I'm printing with PETG and from Orca it is configured to have 70% auxiliary fan (the back fan) for the whole print, but it does not start spinning unless I manually check it.

Could it that be related to the chamber temperature that I see in mainsail? I mean, maybe the commands from the print are setting the right value but to "sustain" the temperature they get turned off?

![Screenshot_20240801_175023](https://github.com/user-attachments/assets/1bebc71e-f8eb-4141-9d1c-1d6750502139)


Also, maybe I'm understanding it wrong fan2 is the auxiliary fan (side one) and fan1 is the back fan? or is it the other way around?

**Reply by ronoaldo** [reply]
Certainly I forgot to RTFM:
https://guilouz.github.io/Creality-Helper-Script-Wiki/helper-script/fans-control-macros/

There is even a print screen of what need to be enabled on Orca for the controls to actually work! I still wonder if the temperature trigger can somewhat mess up with the M106 commands for fan control.

**Comment by SPACEMONKEY-01** [comment]
@MakManIfest I set the min value for Fan 2 and all other fans to 0 and now it's working correctly with Orca when I set the % values. Def should be fixed in the firmware, but oh well.
